### PR TITLE
Fix Withdrawing if inventory is full.

### PR DESCRIPTION
**Pull request by Oribuin**
Fixed the bug reported in #402

### CODE REVIEW

**Comment by Oribuin** [comment]
^ Issues with merging will be because savage labs hasn't gotten it shit together

File: src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java [hunk 1]
```diff
@@ -108,6 +104,7 @@ public void buy(Player player, ShopItem shopItem, int amount) {
         }
 
         if (shopItem.command == null) {
+
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java:107]:
```suggestion
```

File: src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java [hunk 2]
```diff
@@ -121,9 +118,10 @@ public void buy(Player player, ShopItem shopItem, int amount) {
                 itemMeta.setDisplayName(StringUtils.color(shopItem.displayName));
                 itemStack.setItemMeta(itemMeta);
             }
-            
+
             player.getInventory().addItem(itemStack);
         } else {
+
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java:124]:
```suggestion
```

File: src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java [hunk 3]
```diff
@@ -16,28 +14,57 @@
...
 /**
  * Various utils for working with {@link Player}'s.
  */
 public class PlayerUtils {
 
     public static boolean pay(@NotNull Player player, @NotNull Island island, int crystals, double money) {
-        IslandBank islandCrystals = IridiumSkyblock.getInstance().getIslandManager().getIslandBank(island,
+
+        // Don't withdraw stuff if they can't purchase it.
+        if (!canPurchase(player, island, crystals, money))
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java:28]: Either add these curly brackets or write it on one line

File: src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java [hunk 4]
```diff
@@ -60,7 +87,8 @@ public static void sendBorder(@NotNull Player player, @NotNull Island island) {
 
     public static void teleportSpawn(Player player) {
         World spawnWorld = Bukkit.getWorld(IridiumSkyblock.getInstance().getConfiguration().spawnWorldName);
-        if (spawnWorld == null) spawnWorld = Bukkit.getWorlds().get(0);
+        if (spawnWorld == null)
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java:90]: Same here

**Review by dlsf — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java [hunk 5]
```diff
@@ -93,13 +93,9 @@ public void buy(Player player, ShopItem shopItem, int amount) {
         BuyCost buyCost = shopItem.buyCost;
         double vaultCost = calculateCost(amount, shopItem.defaultAmount, buyCost.vault);
         int crystalCost = (int) calculateCost(amount, shopItem.defaultAmount, buyCost.crystals);
+        final Island island = IridiumSkyblockAPI.getInstance().getUser(player).getIsland().get();
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java:96]: Use IridiumSkyblock#getUserManager Also island can be empty you're getting it while not checked

**Review by MertUnverdi — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java (hunk 5 above)
Comment (Oribuin) [reply src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java:96]: Don't blame me, That was in there before I edited it and I also don't believe a player can open the shop if they don't have an Island

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java]: u're just idiot!
may they can get kicked island while they're viewing shop ??

File: src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java (hunk 4 above)
Comment (Oribuin) [reply src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java:90]: My code still forces it onto the next line but sure I _guess_

File: src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java (hunk 5 above)
Comment (Oribuin) [reply src/main/java/com/iridium/iridiumskyblock/shop/ShopManager.java]: Okay?? It's literally your code you baboon, I wouldn't have done it the way it is, Your base code not mine you fool

**Comment by PeachesMLG** [comment]
> ^ Issues with merging will be because savage labs hasn't gotten it shit together

nah thats kinda my fault, Savage quit so obs he aint paying for his repo anymore, so I gotta make my own

**Comment by PeachesMLG** [comment]
Welp Das is the one who made the shop so ima just let him deal with this pr

**Comment by PeachesMLG** [comment]
Btw you know Inventory#addItem returns a hashmap of items it couldnt add (e.g. if theres not enough room) so you could always just make them drop onto the floor?

**Comment by Oribuin** [comment]
> Btw you know Inventory#addItem returns a hashmap of items it couldnt add (e.g. if theres not enough room) so you could always just make them drop onto the floor?

You could do that but when buying items from a shop you don’t want them to be dropped and other players to pick them up, unless someone else wants to make it so only the player that bought it can pick it up. Seems unnecessary

**Comment by Oribuin** [comment]
> > ^ Issues with merging will be because savage labs hasn't gotten it shit together
> 
> nah thats kinda my fault, Savage quit so obs he aint paying for his repo anymore, so I gotta make my own

I’d personally just use rosewood repo, I’d ask Nicole if you could use that if you didn’t want to set up your own

**Review by dlsf — APPROVED** [review]

**Pull request merged**